src/ui/AdvancedSearch/DateInput/DateInput.ts

```diff
@@ -35,16 +36,36 @@ export class DateInput implements IAdvancedSearchInput {
   }
 
   public updateQuery(queryBuilder: QueryBuilder) {
-    let value = this.getValue();
-    if (value) {
-      queryBuilder.advancedExpression.add(this.getValue());
+    try {
+      let value = this.getValue();
+      if (value) {
+        queryBuilder.advancedExpression.add(this.getValue());
+      }
+      this.hideError();
+    } catch (error) {
+      this.showError(error);
     }
   }
 
   protected getRadio(): HTMLInputElement {
     return <HTMLInputElement>$$(this.element).find('input');
   }
 
+
+  private showError(message: string) {
+    this.hideError();
```

Comment:
Weird to see `this.hideError()` as the first line of `showError`. Should probably be `removeError()` :)